### PR TITLE
Add tests for GetEnumerator returning a cloneable enumerator in System.Collections.NonGeneric

### PR DESCRIPTION
**Pull request by hughbe**
Increases code coverage for these types, these are new .NET standard 2.0 additions that were never tested

### CODE REVIEW

File: src/System.Collections.NonGeneric/tests/QueueTests.cs [hunk 1]
```diff
@@ -479,6 +479,22 @@ public static void GetEnumerator(int count)
...
+
+            IEnumerator enumerator = queue.GetEnumerator();
+            ICloneable cloneableEnumerator = (ICloneable)enumerator;
+
+            IEnumerator clonedEnumerator = (IEnumerator)cloneableEnumerator.Clone();
+            Assert.NotSame(enumerator, clonedEnumerator);
+
+            // Cloned and original enumerators should enumerate separately.
+            while (enumerator.MoveNext()) ;
+            Assert.True(clonedEnumerator.MoveNext());
```
Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/QueueTests.cs]: It'd be good to verify as well that:
a) the sequence matches between the original and cloned enumerators
b) that the cloned enumerator starts at the point in the sequence where the clone was done, not necessarily at the start

Comment (stephentoub) [inline src/System.Collections.NonGeneric/tests/QueueTests.cs]: (Same comment applies to the other enumerators

Comment (hughbe) [inline src/System.Collections.NonGeneric/tests/QueueTests.cs]: Cheers, I've responded to your Pr feedback (made more sense in a squashed commit as most code is different)

**Comment by stephentoub** [comment]
@dotnet-bot test Innerloop Ubuntu14.04 Debug Build and Test please

**Pull request merged**